tracks/bash/exercises/hamming/mentoring.md

```diff
@@ -0,0 +1,147 @@
+### Reasonable Solutions
+
+```bash
+#!/usr/bin/env bash
+main () {
+  (( "$#" == 2 )) || {
+      echo "Usage: hamming.sh <string1> <string2>"
+      exit 1
+  }
+  
+  # Regular vars are easier to read when doing fancy parameter expansion.
+  a=$1 b=$2 
+  
+  (( "${#a}" == "${#b}" )) || {
+      echo "left and right strands must be of equal length"
+      exit 1
+  }
+
+  declare -i count
+  for (( i = 0; i < "${#a}"; i++ )); do
+    [[ "${a:i:1}" == "${b:i:1}" ]] || count+=1
+  done
+
+  printf '%d' "$count"
+}
+
+main "$@"
+```
+```bash
+#!/usr/bin/env bash
+main () {
+  (( $# != 2 )) && 
+    echo "Usage: hamming.sh <string1> <string2>" && 
+    exit 1
+  (( ${#1} != ${#2} )) && 
+    echo "left and right strands must be of equal length" && 
+    exit 1
+
+  hamming=0;
+  for (( i = 0; i < ${#1}; i++ )); do
+    [[ "${1:i:1}" != "${2:i:1}" ]] && ((++hamming))
+  done
+
+  echo "$hamming"
+}
+main "$@"
+```
+```bash
+#!/usr/bin/env bash
+main () {
+    if (( $# != 2 )); then 
+        echo "Usage: hamming.sh <string1> <string2>";
```

Comment:
Inconsistent use of trailing `;`. Some statements got them, others don't. Convention is to drop them.